Monika After Story/game/script-python.rpy

```diff
@@ -0,0 +1,1205 @@
+            None
+        )
+
+        if tip_ev is None:
+            return False
+
+        # otherwise, unlocked date is our key
+        if tip_ev.unlock_date is None or tip_ev.shown_count == 0:
+            return False
+
+        # now check the actual day
+        return (
+            datetime.date.today() - tip_ev.unlock_date.date() 
+            >= datetime.timedelta(days=1)
+        )
+
+    def has_day_past_tips(*tip_nums):
+        """
+        Variant of has_day_past_tip that can check multiple numbers
+
+        SEE has_day_past_tip for more info
+
+        RETURNS:
+            true if all the given tip nums have been see nand a day has past
+                since the latest one was unlocked, False otherwise
+        """
+        for tip_num in tip_nums:
+            if not has_day_past_tip(tip_num):
+                return False
+
+        return True
+
+
+# The initial event is getting Monika to talk about python
+# this must be hidden after it has been completed
+init 5 python:
+    addEvent(
+        Event(
+            persistent.event_database,
+            eventlabel="monika_ptod_tip000",
+            category=["python tips"],
+            prompt="Can you teach me about Python?",
+            pool=True
+        )
+    )
+
+label monika_ptod_tip000:
+    m 3eub "You want to learn about Python?"
+    m 3hub "I'm glad!"
+    m 1lksdlb "I don't know {i}that{/i} much about programming, but I will try my best!"
+    m 1esa "Let's start with what Python even is."
+
+    $ hideEventLabel("monika_ptod_tip000", depool=True)
+    jump monika_ptod_tip001
+
+###############################################################################
+init 5 python:
+    addEvent(
+        Event(
+            persistent.event_database,
+            eventlabel="monika_ptod_tip001",
+            category=["python tips"],
+            prompt="What is Python?"
+        )
+    )
+
+label monika_ptod_tip001:
+    # first enable this event
+    $ tip_ev = mas_getEV("monika_ptod_tip001")
+    $ tip_ev.pool = True
+    $ tip_ev.unlocked = True
+
+    m 1esa "Python was created by Guido Van Rossum in the early 90s."
+    m "It is used in many places, including web apps, embedded systems, Linux, and of course..."
+    m 1hua "This mod!"
+    m 1eua "DDLC in particular uses a game engine called RenPy,{w} which is built off of Python."
+    m 3eub "That means if you learn a bit of Python, you can add content to my world!"
+    show monika 5eua with dissolve
+    m "Wouldn't that be great, [player]?"
+    
+    m 4eub "Anyway, I need to mention that there are currently two main versions of Python:{w} Python2 and Python3."
+    m 3eua "These versions are {b}incompatible{/b} with each other because the changes added in Python3 fixed many fundamental design flaws in Python2."
+    m "Even though this introduced a split in the Python community,{w} it's generally agreed that both versions of the language have their own strengths and weaknesses."
```

Comment:
"Even though this caused a rift in the Python community, {w} it's generally agreed upon...